From 9ed9b10641cca1371321c5588ed6648cfd456734 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:00:02 -0400
Subject: [acpi] Drop ACPI bus notifier call chain

Message-id: <20140710010002.22146.68959.stgit@gir.stowe>
Patchwork-id: 85556
O-Subject: [RHEL7.1 PATCH 017/105] ACPI: Drop ACPI bus notifier call chain
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: f716fc2ac037c45a6c641eb9f20ec602e8d04e14

commit f716fc2ac037c45a6c641eb9f20ec602e8d04e14
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Sun Jun 30 23:49:55 2013 +0200

    ACPI: Drop ACPI bus notifier call chain

    There are no users of the ACPI bus notifier call chain,
    acpi_bus_notify_list, any more, so drop it.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/bus.c b/drivers/acpi/bus.c
index 74df4d7..24a2445 100644
--- a/drivers/acpi/bus.c
+++ b/drivers/acpi/bus.c
@@ -499,19 +499,6 @@ static void acpi_bus_check_scope(acpi_handle handle)
   */
 }
 
-static BLOCKING_NOTIFIER_HEAD(acpi_bus_notify_list);
-int register_acpi_bus_notifier(struct notifier_block *nb)
-{
- return blocking_notifier_chain_register(&acpi_bus_notify_list, nb);
-}
-EXPORT_SYMBOL_GPL(register_acpi_bus_notifier);
-
-void unregister_acpi_bus_notifier(struct notifier_block *nb)
-{
- blocking_notifier_chain_unregister(&acpi_bus_notify_list, nb);
-}
-EXPORT_SYMBOL_GPL(unregister_acpi_bus_notifier);
-
 /**
  * acpi_bus_notify
  * ---------------
@@ -525,9 +512,6 @@ static void acpi_bus_notify(acpi_handle handle, u32 type, void *data)
  ACPI_DEBUG_PRINT((ACPI_DB_INFO, "Notification %#02x to handle %p\n",
      type, handle));
 
- blocking_notifier_call_chain(&acpi_bus_notify_list,
-  type, (void *)handle);
-
  switch (type) {
 
  case ACPI_NOTIFY_BUS_CHECK:
diff --git a/include/acpi/acpi_bus.h b/include/acpi/acpi_bus.h
index 365c069..3fb25ec 100644
--- a/include/acpi/acpi_bus.h
+++ b/include/acpi/acpi_bus.h
@@ -363,8 +363,6 @@ extern int acpi_notifier_call_chain(struct acpi_device *, u32, u32);
 extern int register_acpi_notifier(struct notifier_block *);
 extern int unregister_acpi_notifier(struct notifier_block *);
 
-extern int register_acpi_bus_notifier(struct notifier_block *nb);
-extern void unregister_acpi_bus_notifier(struct notifier_block *nb);
 /*
  * External Functions
  */
-- 
1.7.1